frontend/src/pages/Admin/Resources/index.tsx

```diff
@@ -134,27 +110,30 @@ export const Resources: FC = () => {
       >
         <>
           {!editMode ? (
-            <ReadForm fields={readFields} />
+            <ReadForm fields={readFields(resourcesQuery)} />
           ) : (
             <>
               {"new" === formType ? (
                 <CreateForm
                   initialValues={createFormInitialValues}
+                  /* eslint "no-warning-comments": [1, { "terms": ["todo", "fixme"] }] */
                   // todo: Define a good type. "Any" type temporarily permitted.
                   // eslint-disable-next-line @typescript-eslint/no-explicit-any
                   onSubmit={async (values: any) => {
                     return handlePost({
                       formValues: values,
                       apiUrl: `/resources`,
                       handleEditMode: handleEditMode,
-                      queryKeys: [`"/resources/${resourcesQuery?.data?.id}"`],
+                      queryKeys: [`"/resources"`],
                     });
                   }}
-                  editFields={editFields}
+                  editFields={editFields()}
```

Comment:
Just curious why this is a function and not just an array of objects? Is it a constraint of the form library that's being used?

Reply:
this was a bit of copy paste from Adam, but i think his logic was that the readfields requires a param to be passed, and this might be the case for this one at some point, as he is using react query to get the results.